CMakeLists.txt

```diff
@@ -56,9 +56,14 @@ else()
     unset(COMPILER_SUPPORTS_MARCH_NATIVE CACHE)
     CHECK_CXX_COMPILER_FLAG(-march=native COMPILER_SUPPORTS_MARCH_NATIVE)
     if(COMPILER_SUPPORTS_MARCH_NATIVE)
-    	set(CMAKE_CXX_FLAGS "-O3 -march=native -mfma -ffp-contract=fast")
-    elseif(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "arm64")  
-    	set(CMAKE_CXX_FLAGS "-O3 -mcpu=apple-a14 -mfma -ffp-contract=fast")
+        message(STATUS "${CMAKE_HOST_SYSTEM_PROCESSOR}")
+        if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "aarch64")
+    	    set(CMAKE_CXX_FLAGS "-O3 -march=native -ffp-contract=fast")
+        else()
+            set(CMAKE_CXX_FLAGS "-O3 -march=native -mfma -ffp-contract=fast")
+        endif()
+    elseif(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "arm64")
+    	set(CMAKE_CXX_FLAGS "-O3 -mcpu=apple-a14 -ffp-contract=fast")
```

Comment:
why are there two branches here, one for `aarch64` and another for `arm64`?

Reply:
yeah, this is confusing because "arm64" is given on macos, and "aarch64" on linux on the same CPU.

Reply:
The clang compiler on Mac (v15.x) now supports march=native. So we need to change the conditions to reflect this. I'll adjust the conditions and push them.

Reply:
I'd recommend to go with the below if else conditions. 
```
if(COMPILER_SUPPORTS_MARCH_NATIVE)
    set(CMAKE_CXX_FLAGS "-O3 -march=native -ffp-contract=fast")
elseif(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "arm64")  
    set(CMAKE_CXX_FLAGS "-O3 -mcpu=apple-a14 -ffp-contract=fast")
```

Reply:
@mbkumar I did a few more changes as you hinted. I kept the -mfma option in the “old clang” branch but skipped it otherwise where it should be anyway added due to the other flags. Also, there’s a fallback case in the end. Can we merge this like it is now?